### PR TITLE
[BUGFIX][BREAKING] Convert GE version key from great_expectations.__version__ to great_expectations_version

### PR DESCRIPTION
**Pull request by cwerner**
This PR adds the proposed change of the GE version key from great_expectations.__version__ to great_expectations_version to avoid problems with libraries that do not allow dots in dict keys (i.e. python-benedict) (see issue #1564).

Note, this is a simple replacement in the entire codebase. I'm not sure if there are safeguards required to ingest json from older installs? 

Changes proposed in this pull request:
- change of the GE version key from great_expectations.__version__ to great_expectations_version

### CODE REVIEW

File: great_expectations/jupyter_ux/expectation_explorer.py [hunk 1]
```diff
@@ -1939,7 +1939,7 @@ def generate_expectation_suite_editor_widgets(self, data_asset, expectation_suit
         data_asset_state = self.state["data_assets"].get(data_asset_name, {})
         data_asset_expectations = data_asset_state.get("expectations", {})
 
-        ge_version = expectation_suite.get("meta")["great_expectations.__version__"]
+        ge_version = expectation_suite.get("meta")["great_expectations_version"]
```
Comment (jcampbell) [inline great_expectations/jupyter_ux/expectation_explorer.py]: Mark with a todo to return and fix later, but for now let's accept both versions so this isn't breaking.

File: great_expectations/data_asset/data_asset.py [hunk 2]
```diff
@@ -934,15 +934,12 @@ def validate(
 
             # Warn if our version is different from the version in the configuration
             try:
-                if (
-                    expectation_suite.meta["great_expectations.__version__"]
-                    != ge_version
-                ):
+                if expectation_suite.meta["great_expectations_version"] != ge_version:
```
Comment (jcampbell) [inline great_expectations/data_asset/data_asset.py]: Mark with a todo to return and fix later, but for now let's accept both versions so this isn't breaking.

**Review by jcampbell — CHANGES_REQUESTED** [review]
Minor changes to make this non-breaking, but love it -- thank you!

File: great_expectations/data_asset/data_asset.py (hunk 2 above)
Comment (cwerner) [reply great_expectations/data_asset/data_asset.py]: Hi @jcampbell Sorry, I'm not clear if I should change stuff now? Should I just revert these changes at the specified locations and add a comment?

Comment (spbail) [reply great_expectations/data_asset/data_asset.py]: Just came across this :) I think this meant allow both `great_expectations_version` and `great_expectations.__version__` to make the change non-breaking? And add a TODO to clean up later.

File: great_expectations/data_asset/data_asset.py [hunk 3]
```diff
@@ -867,20 +867,19 @@ def validate(
...
-                    expectation_suite.meta["great_expectations.__version__"]
-                    != ge_version
-                ):
+            # TODO: Deprecate "great_expectations.__version__"
+            suite_ge_version = expectation_suite.meta.get(
+                "great_expectations_version"
+            ) or expectation_suite.meta.get("great_expectations.__version__")
+
+            if suite_ge_version:
+                if suite_ge_version != ge_version:
```
Comment (roblim) [inline great_expectations/data_asset/data_asset.py:876]: Made non-breaking

File: great_expectations/jupyter_ux/expectation_explorer.py [hunk 4]
```diff
@@ -1939,7 +1939,10 @@ def generate_expectation_suite_editor_widgets(self, data_asset, expectation_suit
         data_asset_state = self.state["data_assets"].get(data_asset_name, {})
         data_asset_expectations = data_asset_state.get("expectations", {})
 
-        ge_version = expectation_suite.get("meta")["great_expectations.__version__"]
+        # TODO: Deprecate "great_expectations.__version__"
+        ge_version = expectation_suite.get("meta").get(
+            "great_expectations_version"
+        ) or expectation_suite.get("meta").get("great_expectations.__version__")
         column_count = len(column_names)
```
Comment (roblim) [inline great_expectations/jupyter_ux/expectation_explorer.py:1946]: Made non-breaking

File: great_expectations/render/renderer/page_renderer.py [hunk 5]
```diff
@@ -285,7 +285,10 @@ def _render_validation_info(cls, validation_results):
         elif isinstance(run_id, RunIdentifier):
             run_name = run_id.run_name or "__none__"
             run_time = run_id.run_time.strftime("%Y-%m-%dT%H:%M:%S.%fZ")
-        ge_version = validation_results.meta["great_expectations.__version__"]
+        # TODO: Deprecate "great_expectations.__version__"
+        ge_version = validation_results.meta.get(
+            "great_expectations_version"
+        ) or validation_results.meta.get("great_expectations.__version__")
```
Comment (roblim) [inline great_expectations/render/renderer/page_renderer.py:292]: Made non-breaking

File: great_expectations/render/renderer/page_renderer.py [hunk 6]
```diff
@@ -574,7 +577,10 @@ def _render_expectation_suite_header(cls):
     @classmethod
     def _render_expectation_suite_info(cls, expectations):
         expectation_suite_name = expectations.expectation_suite_name
-        ge_version = expectations.meta["great_expectations.__version__"]
+        # TODO: Deprecate "great_expectations.__version__"
+        ge_version = expectations.meta.get(
+            "great_expectations_version"
+        ) or expectations.meta.get("great_expectations.__version__")
```
Comment (roblim) [inline great_expectations/render/renderer/page_renderer.py:583]: Made non-breaking

**Review by roblim — APPROVED** [review]
LGTM! @jcampbell - made changes to make non-breaking.

**Review by jcampbell — APPROVED** [review]

**Pull request merged**